app/mailers/placements/provider_user_mailer.rb

```diff
@@ -26,88 +26,66 @@ def user_membership_destroyed_notification(user, organisation)
       body: t(
-        ".body.#{partner_class}",
+        ".body",
         user_name: user.full_name,
         source_organisation: source_organisation.name,
         partner_organisation: partner_organisation.name,
-        link:,
+        link: placements_provider_partner_schools_url(partner_organisation),
       ),
     )
   end
 
   def partnership_destroyed_notification(user, source_organisation, partner_organisation)
-    if partner_organisation.is_a?(Provider)
-      partner_class = "provider"
-      link = placements_provider_partner_schools_url(partner_organisation)
-    elsif partner_organisation.is_a?(School)
-      partner_class = "school"
-      link = placements_school_partner_providers_url(partner_organisation)
-    else
-      raise InvalidOrganisationError, "#partner_organisation must be either a Provider or School"
-    end
-
     notify_email(
       to: user.email,
-      subject: t(".subject.#{partner_class}", organisation: partner_organisation.name),
+      subject: t(".subject", organisation: partner_organisation.name),
       body: t(
-        ".body.#{partner_class}",
+        ".body",
         user_name: user.full_name,
         source_organisation: source_organisation.name,
         partner_organisation: partner_organisation.name,
-        link:,
+        link: placements_provider_partner_schools_url(partner_organisation),
       ),
     )
   end
 
   def placement_provider_assigned_notification(user, provider, placement)
-    placement_link = placements_provider_placement_url(provider, placement)
-    school = placement.school
-
     notify_email(
       to: user.email,
-      subject: t(".subject", school_name: school.name),
+      subject: t(".subject", school_name: school(placement).name),
       body: t(
         ".body",
         provider_name: provider.name,
         placement_name: placement.decorate.title,
-        school_name: school.name,
-        school_email: school.school_contact.email_address,
-        link: placement_link,
+        school_name: school(placement).name,
+        school_email: school(placement).school_contact.email_address,
+        link: placements_provider_placement_url(provider, placement),
       ),
     )
   end
 
   def placement_provider_removed_notification(user, provider, placement)
-    placement_link = placements_provider_placement_url(provider, placement)
-    school = placement.school
-
     notify_email(
       to: user.email,
-      subject: t(".subject", school_name: school.name),
+      subject: t(".subject", school_name: school(placement).name),
       body: t(
         ".body",
         provider_name: provider.name,
         placement_name: placement.decorate.title,
-        school_name: school.name,
-        school_email: school.school_contact.email_address,
-        link: placement_link,
+        school_name: school(placement).name,
+        school_email: school(placement).school_contact.email_address,
```

Comment:
I feel like this is probably a more straightforward way to do the same thing?

```suggestion
        school_name: placement.school.name,
        school_email: placement.school.school_contact.email_address,
```

While it's not always good to rely on deeply nested objects (i.e. [The Law of Demeter](https://thoughtbot.com/ruby-science/law-of-demeter.html)), I think we're already quite far down that path. In this case it feels like a lesser evil compared to the indirection introduced by of the private `school(placement)` method.